src/AndersonAcceleration.f90

```diff
@@ -85,109 +85,56 @@ MODULE AndersonAccelerationTypes
 
   IF(.NOT. solver%isInit) THEN
-    IF(n < 1) THEN
-      CALL eAndersonAccelerationType%raiseError('Incorrect input to '// &
-          modName//'::'//myName//' - Number of values (n) must be '// &
-          'greater than 0!')
-    ELSE
-      solver%n=n
-    ENDIF
-
-    IF((nlocal < 1) .AND. (nlocal > n)) THEN
-      CALL eAndersonAccelerationType%raiseError('Incorrect input to '// &
-          modName//'::'//myName//' - Number of values (nlocal) must be '// &
-          'greater than 0 and less than or equal to (n)!')
+    IF(solver%N < 1) THEN
+      CALL eAndersonAccelerationType%raiseError('Incorrect input to '//modName// &
+          '::'//myName//' - Number of unkowns (N) must be greater than 0!')
     ENDIF
 
-    IF(depth < 0) THEN
-      CALL eAndersonAccelerationType%raiseError('Incorrect input to '// &
-          modName//'::'//myName//' - Depth values (depth) must be '// &
-          'greater than or equal to 0!')
-    ELSE
-      solver%depth=depth
-    ENDIF
-
-    IF(start <= 0) THEN
-      CALL eAndersonAccelerationType%raiseError('Incorrect input to '// &
-          modName//'::'//myName//' - Starting iteration must be '// &
-          'greater than to 0!')
-    ENDIF
-
-    IF((beta<=0.0_SRK) .OR. (beta>1.0_SRK)) THEN
-      CALL eAndersonAccelerationType%raiseError('Incorrect input to '// &
-          modName//'::'//myName//' - Beta must be '// &
-          'greater than 0 and less than or equal to 1!')
-    ELSE
-      solver%beta=beta
-    ENDIF
-
-#ifdef FUTILITY_HAVE_Trilinos_NOX
-    CALL tmpPL%clear()
-    CALL tmpPL%add('VectorType->n',n)
-    CALL tmpPL%add('VectorType->MPI_Comm_ID',solver%MPIparallelEnv%comm)
-    CALL tmpPL%add('VectorType->nlocal',nlocal)
-    CALL tmpPL%add('VectorType->engine',VM_TRILINOS)
-    CALL VectorFactory(solver%X,tmpPL)
-    CALL solver%X%set(0.0_SRK)
-
-    SELECTTYPE(x=>solver%X); TYPE IS(TrilinosVectorType)
-      CALL Anderson_Init(solver%id,solver%depth,solver%beta,start,x%b)
-    ENDSELECT
-#else
-    CALL eAndersonAccelerationType%raiseError(modName//'::'//myName// &
-        ' - NOX (Trilinos) is not present in build')
-#endif
+    !Allocate member arrays
+    M=solver%depth+1
+    ALLOCATE(solver%x(solver%N,M),solver%Gx(solver%N,M),solver%r(solver%N,M), &
+        solver%A(solver%depth,solver%depth),solver%alpha(M),solver%b(solver%depth))
+    solver%x(:,:)=0.0_SRK
+    solver%Gx(:,:)=0.0_SRK
+    solver%r(:,:)=0.0_SRK
+    solver%A(:,:)=0.0_SRK
+    DO i=1,solver%depth
+      solver%A(i,i)=1.0_SRK
+    ENDDO
+    solver%alpha(:)=0.0_SRK
+    solver%b(:)=0.0_SRK
+    solver%s=0
     solver%isInit=.TRUE.
   ELSE
-    CALL eAndersonAccelerationType%raiseError('Incorrect call to '// &
-        modName//'::'//myName//' - AndersonAccelerationType already initialized')
+    CALL eAndersonAccelerationType%raiseError('Incorrect call to '//modName// &
+        '::'//myName//' - TrilinosAndersonAccelerationType already initialized')
```

Comment:
whaaa?

Reply:
Hah, woops. Leftover from when I was trying to split it to keep both.